### ISSUE TITLE
Some help to have webubbub running on local

### ISSUE THREAD

**Issue by jrouaix** [flusio/Webubbub#8]
Hello @marienfressinaud

I'm trying to have a websub hub running locally to make some tests, and your project seems to be the most maintained around here 👍🏼  congrats !

Could you help me one this one ?
My Dockerfile is 
```dockerfile
FROM php

# prepare build dir
WORKDIR /app

RUN \
  apt-get update && apt-get install -y git && \
  git clone https://github.com/flusio/Webubbub.git . && \
  git submodule update --init --recursive && \
  cp env.sample .env

RUN \
  php ./webubbub --request /system/init && \
  php ./webubbub --request /system/migrate
# run the app
CMD ["php", "./webubbub", "--request", "/"]
```

Now what should I do ...  `mode` ?

Or perhaps I didn't understand this project
What I'd like is having a local hub, and also be able to subscribe to it to test my `publish` notifications to it 

Any idea ?

**Comment by marienfressinaud** [comment]
Hi @jrouaix 

> I'm trying to have a websub hub running locally to make some tests, and your project seems to be the most maintained around here 👍🏼 congrats !

Ha ha, I didn't work on it for several months, but I guess the others are not more maintained :D

To run Webubbub, you'll need to run a web server. The last command of your Dockerfile doesn't run the server. Instead, you should have something like:

```dockerfile
CMD ["php", "-t", "./public/", "-S", "127.0.0.1:8000", "./public/index.php"]
```

but I'm not even sure it would work well.

Instead, you can follow the documentation of the development environment: https://github.com/flusio/Webubbub/blob/main/docs/development-install.md

**Comment by jrouaix** [comment]
Well it worked with 

```Dockerfile
CMD ["php", "-t", "./public/", "-S", "0.0.0.0:8000", "./public/index.php"]
```

thanks 🎉 

So know I just have to run commands in the running container to `subscribe` to a topic and I should see stuff incomming when my app notifies the hub ?

**Comment by marienfressinaud** [comment]
You can either subscribe with the CLI, or call the endpoint externally. For instance, with Curl:

```console
$ curl -d "hub.mode=subscribe&hub.callback=https://example.com/callback&hub.topic=https://example.com/topic" -X POST http://localhost:8000
```

**Comment by marienfressinaud** [comment]
I forgot something! You also have to run the following script at a regular time in the container: https://github.com/flusio/Webubbub/blob/main/bin/jobs.sh

**Comment by jrouaix** [comment]
I might be close to it working but it's not ...

```Dockerfile
FROM php

# prepare build dir
WORKDIR /app

RUN \
  apt-get update && apt-get install -y git cron && \
  git clone https://github.com/flusio/Webubbub.git . && \
  git submodule update --init --recursive && \
  cp env.sample .env

RUN \
  php ./webubbub --request /system/init && \
  php ./webubbub --request /system/migrate

ADD cron.job /opt/webubbub/cron.cronjob
RUN \
  chmod 0644 /opt/webubbub/cron.cronjob && \
  crontab /opt/webubbub/cron.cronjob

# run the app
# CMD ["php", "./webubbub", "--request", "/"]
CMD (cron -f &) && php -t ./public/ -S "0.0.0.0:8000" "./public/index.php"
```

And then usage : 
```sh
curl -d "hub.mode=publish&hub.topic=http://localhost:8080/region/Nouvelle-Aquitaine/75/feed" -X POST http://localhost:8000/
```

first try returns :
```sh
<br />
<b>Deprecated</b>:  Creation of dynamic property Minz\Database::$connect_to_db is deprecated in <b>/app/lib/Minz/src/Database.php</b> on line <b>33</b><br />
<br />
<b>Deprecated</b>:  Creation of dynamic property Webubbub\models\Content::$id is deprecated in <b>/app/lib/Minz/src/Model.php</b> on line <b>195</b><br />
<br />
<b>Deprecated</b>:  Creation of dynamic property Webubbub\models\Content::$created_at is deprecated in <b>/app/lib/Minz/src/Model.php</b> on line <b>195</b><br />
<br />
<b>Deprecated</b>:  Creation of dynamic property Webubbub\models\Content::$fetched_at is deprecated in <b>/app/lib/Minz/src/Model.php</b> on line <b>195</b><br />
<br />
<b>Deprecated</b>:  Creation of dynamic property Webubbub\models\Content::$status is deprecated in <b>/app/lib/Minz/src/Model.php</b> on line <b>195</b><br />
<br />
<b>Deprecated</b>:  Creation of dynamic property Webubbub\models\Content::$url is deprecated in <b>/app/lib/Minz/src/Model.php</b> on line <b>195</b><br />
<br />
<b>Deprecated</b>:  Creation of dynamic property Webubbub\models\Content::$links is deprecated in <b>/app/lib/Minz/src/Model.php</b> on line <b>195</b><br />
<br />
<b>Deprecated</b>:  Creation of dynamic property Webubbub\models\Content::$type is deprecated in <b>/app/lib/Minz/src/Model.php</b> on line <b>195</b><br />
<br />
<b>Deprecated</b>:  Creation of dynamic property Webubbub\models\Content::$content is deprecated in <b>/app/lib/Minz/src/Model.php</b> on line <b>195</b><br />
<br />
<b>Warning</b>:  Cannot modify header information - headers already sent by (output started at /app/lib/Minz/src/Database.php:33) in <b>/app/public/index.php</b> on line <b>25</b><br />
<br />
<b>Warning</b>:  Cannot modify header information - headers already sent by (output started at /app/lib/Minz/src/Database.php:33) in <b>/app/public/index.php</b> on line <b>25</b><br />
```

and then always returns : 
```sh
<br />
<b>Deprecated</b>:  Creation of dynamic property Minz\Database::$connect_to_db is deprecated in <b>/app/lib/Minz/src/Database.php</b> on line <b>33</b><br />
<br />
<b>Warning</b>:  Cannot modify header information - headers already sent by (output started at /app/lib/Minz/src/Database.php:33) in <b>/app/public/index.php</b> on line <b>25</b><br />
<br />
<b>Warning</b>:  Cannot modify header information - headers already sent by (output started at /app/lib/Minz/src/Database.php:33) in <b>/app/public/index.php</b> on line <b>25</b><br />
```

**Comment by marienfressinaud** [comment]
This is an issue with PHP 8.2, you should try `FROM php:8.0` (or `FROM php:8.1` but it's not tested so you may have problems with this version as well).

**Comment by marienfressinaud** [comment]
Hello @jrouaix, just to let you know that I just released a big update (nothing fancy but a lot of small breaking changes.
I tried to document everything in the [changelog](https://github.com/flusio/Webubbub/blob/main/CHANGELOG.md#migration-notes).

TL;DR

- Webubbub now supports PHP 8.0+ (including 8.2)
- the PHP extension `pcntl` is required
- I've changed the format of the name of the migrations so you may have to update the file `data/migrations_version.txt` (or you can reset the database if you have nothing important)
- the CLI commands changed. For instance, to setup or migrate the database, you'll have to run `php ./cli migrations setup`
- I changed the script `bin/jobs.sh` to use a more robust system. In your case you can continue to use a cron job with the following command:

```cron
* * * * * php /var/www/Webubbub/cli jobs watch --stop-after=5
```

Let me know if you need help!
